### PR TITLE
gnome3.at-spi2-core: Set environment variable `NO_AT_BRIDGE=1` if disabled

### PR DESCRIPTION
**Pull request by fmthoma**
As suggested by @jtojnar and @abbradar in #16327.

###### Motivation for this change
See https://github.com/NixOS/nixpkgs/issues/16327#issuecomment-315729994

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: nixos/modules/services/desktops/gnome3/at-spi2-core.nix [hunk 1]
```diff
@@ -28,14 +28,15 @@ with lib;
...
+    mkIf config.services.gnome3.at-spi2-core.enable {
+      environment.systemPackages = [ pkgs.at_spi2_core ];
+      services.dbus.packages = [ pkgs.at_spi2_core ];
+      systemd.packages = [ pkgs.at_spi2_core ];
+    }
+
+    mkIf !config.services.gnome3.at-spi2-core.enable {
+      environment.variables.NO_AT_BRIDGE = "1";
+    }
+  ]
```
Comment (jtojnar) [inline nixos/modules/services/desktops/gnome3/at-spi2-core.nix]: Missing semicolon?

File: nixos/modules/services/desktops/gnome3/at-spi2-core.nix [hunk 2]
```diff
@@ -28,14 +28,15 @@ with lib;
...
-
-    systemd.packages = [ pkgs.at_spi2_core ];
-
-  };
+  config = mkMerge [
+    mkIf config.services.gnome3.at-spi2-core.enable {
+      environment.systemPackages = [ pkgs.at_spi2_core ];
+      services.dbus.packages = [ pkgs.at_spi2_core ];
+      systemd.packages = [ pkgs.at_spi2_core ];
+    };
```
Comment (jtojnar) [inline nixos/modules/services/desktops/gnome3/at-spi2-core.nix]: Only after the `config` attribute, lists items are not supposed to end with semicolons.

**Comment by jtojnar** [comment]
Hmm, the evaluation still fails 
https://gist.github.com/3cb5e6fcc5b61fc40ed9def7e104efb1

**Comment by fmthoma** [comment]
Yeah, the parentheses around the negation were missing...

**Comment by fmthoma** [comment]
Also added the missing parentheses around the two `mkIf … {…}` expressions. Now all checks are passing.

**Review by jtojnar — APPROVED** [review]
I checked both options with xfce and it works correctly. Thanks.

**Pull request merged**